src/transaction/log_recovery.c

```diff
@@ -3319,6 +3334,20 @@ log_recovery_analysis_from_trantable_snapshot (THREAD_ENTRY * thread_p,
   //
   log_Gl.mvcc_table.complete_mvccids_if_still_active (LOG_SYSTEM_TRAN_INDEX, in_gaps_mvccids, false);
 
+  if (!MVCC_ID_PRECEDES (log_rcv_context.get_largest_mvccid (), log_Gl.hdr.mvcc_next_id))
```

Comment:
Nit.
Why not?
I find it more natural this way.
```suggestion
  if (log_Gl.hdr.mvcc_next_id < log_rcv_context.get_largest_mvccid ())
```

Reply:
Even `if (log_Gl.hdr.mvcc_next_id == log_rcv_context.get_larget_mvccid())`,
log_Gl.hdr.mvcc_next_id should be added by 1.

Using macros like MVCC_ID_PRECEDES is also intended to make the code easier to understand, but I used the macro above in case there is a change in the comparison policy for MVCCID values ​​in the future. 
The mvccid values ​​are designed to be assigned circular, but they are not implemented by design, so I think there will be changes in the future.

Reply:
I applied your suggestion when set_largest_mvccid () is called.